### PR TITLE
rockchip: add FriendlyARM NanoPC T4 support

### PR DESCRIPTION
**Pull request by msdos03**
Hardware
--------
RockChip RK3399 ARM64 (6 cores)
4GB LPDDR3 RAM
1x 1000 Base-T
1 GPIO LED (status)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
16GB eMMC
1x USB 3.0 Port
2x USB 2.0 Port
1x USB Type-C Port
1x M.2 PCI-E Port
AP6356S (BCM4356) SDIO WiFi & Bluetooth adapter

### CODE REVIEW

**Comment by msdos03** [comment]
@1715173329

**Comment by 1715173329** [comment]
https://github.com/openwrt/openwrt/blob/e9911f10e482f3174f745a36c0c9fd7964758caf/target/linux/rockchip/patches-6.1/107-arm64-dts-rockchip-Update-LED-properties-for-Orange-.patch
it would be nice to add system-LED aliases for OpenWrt, otherwise LGTM

**Comment by msdos03** [comment]
@1715173329 Done. Is that OK?

**Comment by 1715173329** [comment]
> @1715173329 Done. Is that OK now?

Please split your changes into 2 commits, one for the uboot and another for the target.
Otherwise LGTM

**Comment by msdos03** [comment]
@1715173329 OK, what else should be done before merge?

**Comment by 1715173329** [comment]
> @1715173329 OK, what else should be done before merge?

Just take a rest ;)

**Comment by msdos03** [comment]
@hauke

File: package/boot/uboot-rockchip/patches/105-nanopc-t4-fix-memory-unstability.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+Signed-off-by: Lu jicong <jiconglu58@gmail.com>
+---
+ arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi b/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
+index 17201bcf41..8b6c9059ab 100644
+--- a/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
++++ b/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
```
Comment (hauke) [inline package/boot/uboot-rockchip/patches/105-nanopc-t4-fix-memory-unstability.patch:18]: Please send this patch to the upstream U-Boot mailling list too to get it integrated into upstream U-Boot.

Comment (1715173329) [reply package/boot/uboot-rockchip/patches/105-nanopc-t4-fix-memory-unstability.patch:18]: https://lore.kernel.org/u-boot/20230705135816.182356-1-jiconglu58@gmail.com/

It's already there ;)

**Pull request merged**